src/material.cpp

```diff
@@ -451,6 +451,97 @@ void Material::init_thermal()
   thermal_tables_ = tables;
 }
 
+void Material::collision_stopping_power(double* s_col, bool positron)
+{
+  // Average electron number and average atomic weight
+  double electron_density = 0.0;
+  double mass_density = 0.0;
+
+  // Log of the mean excitation energy of the material
+  double log_I = 0.0;
+
+  // Effective number of conduction electrons in the material
+  double n_conduction = 0.0;
+
+  // Oscillator strength and square of the binding energy for each oscillator
+  // in material
+  std::vector<double> f;
+  std::vector<double> e_b_sq;
+
+  for (int i = 0; i < element_.size(); ++i) {
+    const auto& elm = data::elements[element_[i]];
+    double awr = data::nuclides[nuclide_[i]]->awr_;
+
+    // Get atomic density of nuclide given atom/weight percent
+    double atom_density = (atom_density_[0] > 0.0) ?
+      atom_density_[i] : -atom_density_[i] / awr;
+
+    electron_density += atom_density * elm.Z_;
+    mass_density += atom_density * awr * MASS_NEUTRON;
+    log_I += atom_density * elm.Z_ * std::log(elm.I_);
+
+    for (int j = 0; j < elm.n_electrons_.size(); ++j) {
+      if (elm.n_electrons_[j] < 0) {
+        n_conduction -= elm.n_electrons_[j] * atom_density;
+        continue;
+      }
+      e_b_sq.push_back(elm.ionization_energy_[j] * elm.ionization_energy_[j]);
+      f.push_back(elm.n_electrons_[j] * atom_density);
+    }
+  }
+  log_I /= electron_density;
+  n_conduction /= electron_density;
+  for (auto& f_i : f) f_i /= electron_density;
+
+  // Get density in g/cm^3 if it is given in atom/b-cm
+  double density = (density_ < 0.0) ? -density_ : mass_density / N_AVOGADRO;
+
+  // Calculate the square of the plasma energy
+  double e_p_sq = PLANCK_C * PLANCK_C * PLANCK_C * N_AVOGADRO *
+    electron_density * density / (2.0 * PI * PI * FINE_STRUCTURE *
+    MASS_ELECTRON_EV * mass_density);
+
+  // Get the Sternheimer adjustment factor
+  double rho = sternheimer_adjustment(f, e_b_sq, e_p_sq, n_conduction, log_I,
+    1.0e-6, 100);
+
+  // Classical elctron radius in cm
+  double r_e = PLANCK_C / (2.0e8 * PI * FINE_STRUCTURE * MASS_ELECTRON_EV);
```

Comment:
Consider:
```C++
constexpr double CM_PER_ANGSTROM {1.0e-8};
constexpr double r_e = CM_PER_ANGSTROM * PLANCK_C / (2.0 * PI * FINE_STRUCTURE * MASS_ELECTRON_EV)
```
Compilers will do constant folding, so better to be more expressive here.